test/sub_transport/test_basic.py

```diff
@@ -155,6 +155,12 @@ async def test_handle_local_echo(self, client):
         client.datagram_received(test_data, ("127.0.0.1", 502))
         assert client.recv_buffer == test_data + test_data
         assert not client.sent_buffer
+        client.recv_buffer = b""
```

Comment:
This is a good test, and show that transport do have a problem !

If you change the PR to avoid the future, but only have the changes in transport I am happy to merge it.

Reply:
The future is the main fix for my bug. I need it or another method to accomplish the same result.

There's another way I considered solving the issue: the send_buffer could be appended instead of reset when a new packet is sent. 

I chose to do it this way seeing I felt it would be more robust for every transaction to return when it is completed including receiving the local echo. If we append to send_buffer then the broadcast will still return early and the next transaction will continue before the local echo is received. This seemed more risky to me.